### ISSUE TITLE
Can't run on Ubuntu 20.04

### ISSUE THREAD

**Issue by Deimantas69** [jsoysouvanh/Refureku#8]
When running cmake with the example CMakeLists, I get the following error:
`Bin/RefurekuGenerator: error while loading shared libraries: libclang.so.12: cannot open shared object file: No such file or directory`

cmake file:
```
cmake_minimum_required(VERSION 3.21)
project(RefurekuTest)

set(CMAKE_CXX_STANDARD 17)


add_executable(RefurekuTest main.cpp)
target_compile_features(RefurekuTest PUBLIC cxx_std_17)


target_include_directories(RefurekuTest PUBLIC ${PROJECT_SOURCE_DIR}/Include)
target_link_directories(RefurekuTest PRIVATE ${PROJECT_SOURCE_DIR}/Bin)
target_link_libraries(RefurekuTest PRIVATE Refureku)

add_custom_target(RunRefurekuGenerator
        WORKING_DIRECTORY ${PROJECT_SOURCE_DIR}  #PROJECT_SOURCE_DIR is the path to the directory containing the Cmake with the project command
        COMMAND ${PROJECT_SOURCE_DIR}/Bin/RefurekuGenerator ${PROJECT_SOURCE_DIR}/RefurekuSettings.toml)

add_dependencies(RefurekuTest RunRefurekuGenerator)
```

**Comment by jsoysouvanh** [comment]
Hi,

I'm not very familiar with Linux systems (I don't have a linux machine to test it so I just tested it with the Window's Linux subsystem).
It looks like your system can't locate the libclang.so.12 shared library (which is perfectly normal since it is not installed on your machine) so the generator can't run. You should have a look at [this link; Step 4: Making the library available at runtime](https://www.cprogramming.com/tutorial/shared-libraries-linux-gcc.html) which explains steps very well. Since the generator binaries are prebuilt, you won't be able to use rpath so check directly the LD_LIBRARY_PATH or ldconfig options.
The LD_LIBRARY_PATH is the "quick" option but doesn't sound clean, so you'd better move the libclang.so to a library folder, make a symlink from libclang.so.12 to libclang.so, and try running the generator again.

Roughly something like this (not tested so there may be typo):
```bash
cp RefurekuTest/Bin/libclang.so /usr/lib
chmod 0755 /usr/lib/libclang.so
cd /usr/lib/
ln -s libclang.so libclang.so.12
ldconfig
```

Then check that RefurekuGenerator points correctly to libclang.so.12:
```bash
ldd RefurekuTest/Bin/RefurekuGenerator
```

You should have libclang.so.12 => /usr/lib/libclang.so.12.

The generator should be able to run correctly after that. Let me know if it solves your problem.

**Comment by Deimantas69** [comment]
Thanks for the detailed response, Initially I thought there would've been a way to automate this/link the file using CMake, however this seems to do the trick

**Comment by jsoysouvanh** [comment]
I made some research yesterday and stumbled across the RUNPATH tag.
I will try to add the executable path so that it looks automatically in the executable's directory for the libclang library (just like in Windows).

**Comment by jsoysouvanh** [comment]
I released [a new version (v2.0.2)](https://github.com/jsoysouvanh/Refureku/releases/tag/v2.0.2) with a few updates including the automatic libclang.so lookup as long as it is located next to RefurekuGenerator. Feel free to reopen the issue if something's wrong.
